src/big/big_rational.cr

```diff
@@ -237,7 +263,18 @@ end
 struct Float
   include Comparable(BigRational)
 
+  # Returns a `BigRational` representing this float.
+  def to_big_r
+    BigRational.new(self)
+  end
+
   def <=>(other : BigRational)
     -(other <=> self)
   end
 end
+
+module Math
+  def frexp(value : BigRational)
+    frexp value.to_big_f
```

Comment:
With a float, frexp extracts the exponent and mantissa. These are values from the binary representation of the float. With other types of number encoding, such as a rational, they aren't the fundamental way the number is represented so there could possibly be conversion errors.